src/Configuration/RectorConfigBuilder.php

```diff
@@ -782,6 +786,14 @@ public function withRules(array $rules): self
     {
         $this->rules = array_merge($this->rules, $rules);
 
+        // log all explicitly registered rules
+        // we only check the non-configurable rules, as the configurable ones might override them
+        $nonConfigurableRules = array_filter(
+            $rules,
+            fn (string $rule): bool => ! is_a($rule, ConfigurableRectorInterface::class, true)
+        );
+        SimpleParameterProvider::addParameter(Option::ROOT_STANDALONE_REGISTERED_RULES, $nonConfigurableRules);
```

Comment:
this should be moved to `RectorConfig` `rule` or `rules` method imo, so it will works on both `RectorConfig` and `RectorConfigBuilder`, since `RectorConfigBuilder` consume it.

Reply:
Duplicate to: https://github.com/rectorphp/rector-src/pull/6761#discussion_r1977929063